web/modules/custom/rdf_entity/src/Entity/Query/Sparql/SparqlArg.php

```diff
@@ -41,16 +109,69 @@ public static function uri($uri) {
    *
    * @return string
    *   Sparql escaped string literal.
+   *
+   * @deprecated To be replaced and include language or datatype format.
    */
   public static function literal($value) {
-    // @todo Support all xml data types, as well as language extensions.
-    $matches = 1;
-    while ($matches) {
-      $matches = 0;
-      $value = str_replace('"""', '', $value, $matches);
+    return self::serialize($value, 'literal');
+  }
+
+  /**
+   * Array of Literals Query argument.
+   *
+   * @param array $values
+   *   An array of strings to be escaped.
+   *
+   * @return string
+   *   Sparql escaped string literal.
+   *
+   * @deprecated To be replaced and include language or datatype format.
+   */
+  public static function literals(array $values) {
+    foreach ($values as $index => $value) {
+      // @todo: Avoid recreating the class?
+      $values[$index] = self::serialize($value, 'literal');
     }
+    return $values;
+  }
 
-    return '"""' . $value . '"""';
+  /**
+   * Returns a serialized version of the given value of the given format.
+   *
+   * @param string $value
+   *   The value to be serialized.
+   * @param string $format
+   *   One of the formats defined in
+   *   \Drupal\rdf_entity\RdfFieldHandler::getSupportedDatatypes().
+   * @param string $lang
+   *   The lang code.
+   *
+   * @return string
+   *   The outcome of the serialization.
+   */
+  public static function serialize($value, $format, $lang = NULL) {
+    $data['value'] = $value;
+    switch ($format) {
+      case RdfFieldHandler::RESOURCE:
+        $data['type'] = 'uri';
+        break;
+
+      case RdfFieldHandler::NON_TYPE:
```

Comment:
NON_TYPE is also a type. It is actually a literal. It is just a non-type for us because we do not need special handling. We just need to escape dangerous string parts (we still handle it and pass it through a serializer).